File: CPP/Monitor.cpp
#include "Elektronik.cpp"

class Monitor : public Elektronik {
private:
    //atribut pada monitor
    string ukuran_layar, layar, resolusi;

public:
    //meanggil pada Elektronik 
    Monitor() : Elektronik(), ukuran_layar(""), layar(""), resolusi("") {}
    Monitor(int i, string n, string m, string h, string g, string u, string l, string r)
        : Elektronik(i, n, m, h, g), ukuran_layar(u), layar(l), resolusi(r) {}

    //getter akses data
    string getUkuranLayar() const{ 
        return ukuran_layar; 
    }
    string getLayar() const{ 
        return layar; 
    }
    string getResolusi() const{ 
        return resolusi; 
    }

    //setter ubah data
    void setUkuranLayar(string u){ 
        ukuran_layar = u; 
    }
    void setLayar(string l){ 
        layar = l; 
    }
    void setResolusi(string r){ 
        resolusi = r; 
    }
};

File: PHP/Monitor.php
<?php
require_once "Elektronik.php";

class Monitor extends Elektronik {
    protected $ukuranLayar;
    protected $resolusi;

    public function __construct($id, $nama, $merek, $harga, $gambar, $ukuranLayar, $resolusi) {
        parent::__construct($id, $nama, $merek, $harga, $gambar);
        $this->ukuranLayar = $ukuranLayar;
        $this->resolusi = $resolusi;
    }

    public function getUkuranLayar() { return $this->ukuranLayar; }
    public function getResolusi() { return $this->resolusi; }

    public function setUkuranLayar($ukuranLayar) { $this->ukuranLayar = $ukuranLayar; }
    public function setResolusi($resolusi) { $this->resolusi = $resolusi; }
}
?>
